hypothesis-python/src/hypothesis/strategies/_internal/strategies.py

```diff
@@ -315,25 +315,30 @@ def example(self) -> Ex:
                     "#drawing-interactively-in-tests for more details."
                 )
 
+        try:
+            return self.__examples.pop()
+        except (AttributeError, IndexError):
+            self.__examples: List[Ex] = []
+
         from hypothesis.core import given
 
         # Note: this function has a weird name because it might appear in
         # tracebacks, and we want users to know that they can ignore it.
         @given(self)
         @settings(
             database=None,
-            max_examples=10,
+            max_examples=100,
```

Comment:
When generating complex data, this slows things down 10x.  Had my function go from 30 seconds to 5 minutes due to this change.

Reply:
Unfortunately that's essentially unavoidable in order to get representative examples, because the first ten examples are drawn from a different 'starter' distribution.

I'm sorry to hear that your function is so slow though.  (you're using `.example()` interactively, right?)

Reply:
Appreciate you taking the time to respond. I worked around this now by copy/pasting the given/settings function and changing max_examples to 1.  (I don't care if my example is representative.)

I confess to abusing the system... kind of.  I wrote an internal library on top of hypothesis that takes an OpenApi spec and generates a single "example" of valid data that matches the spec.   My function is meant to be called interactively.

But I have a test suite for that function so it's also getting called that way.

Reply:
Makes sense! I'd also recommend checking out https://github.com/schemathesis/schemathesis

Reply:
Ha that's a great find. Looks like we made very similar tools around the same time! I'll probably deprecate mine. Thanks!